README_Matthew.md

```diff
@@ -0,0 +1,41 @@
+## Question One
+
+- Generate the Terraform templates to import the `annalise-ai-datalake` bucket into Terraform in order to manage it as IaC, including any suggested configuration improvements that could be made to the harrison.ai Terraform templates.
+Add KMS key
+Add S3 bucket-level Public Access Block configuration
+Add Bucket security policies
+Add intelligent_tiering
+
+- Detail how to import the bucket into Terraform.
+Add backend.tf
+terraform init
+terraform import module.s3_bucket.aws_s3_bucket.this[0] annalise-ai-datalake
+terraform plan
+
+- Describe how you would copy the data in `annalise-ai-datalake` to `harrison-ai-landing`.  You may wish to provide Terraform templates for any resources that would be used by this process, but it is not required.
+Create s3 access point and IAM Role on harrison-ai-landing
+Create VPCE and IAM Role on annalise-ai-datalake
+Share harrison-ai-landing kms key with annalise-ai account
+aws sts assume-role —role-are "harrison-ai-landing IAM Role" —role-session-name "Data-copy"
+aws s3 cp s3://SOURCE_BUCKET_NAME/pathxxx/* s3://"s3 access point alias"/pathxxx/* —sse aws:kms —sse-kms-key-id "harrison-ai-landing kms arn"
```

Comment:
Can you suggest any ways to improve the performance and/or reliability of this step when the bucket contains a large number of files?

Reply:
There are multiple ways which we could use to transfer a large number of files between S3 buckets.
AWS datasync,  we could set both source and destination to s3 and also use CloudWatch to monitor the transfer.
AWS S3 batch replication,  we could create a replication rule to copy data and also generate a report.
AWS CLI,  we could use a robust ec2 server to cp data in parallel , data could be split by prefix or folder.